### DISCUSSION TITLE
Offline support on Nuxt - keeps showing "You're offline"

### DISCUSSION THREAD

**Discussion by felixfavour** [nuxt/nuxt#27443] [Questions]
![image](https://github.com/nuxt/nuxt/assets/46944485/e0ab23c6-eb35-4d48-8d30-84fd76be24e6)

Hello :)

I have had an issue with Nuxt offline for a few weeks now.

Whenever my app is offline Nuxt shows this screenshot containing the app favicon and a "You're offline" text, but I would love for it to show my index.vue/homepage.

I've tried multiple things and looked through similar github issues to get it to work but nothing has worked just yet.

Could there be anything I am doing wrong?

I should mention that my app is a PWA, and if there is any additional config you need to see to help out, I'm happy to share.

**Comment by TheAlexLichter** [comment]
You can use `navigator.onLine` or https://vueuse.org/core/useOnline/ + designing the screen on your own

**Reply by felixfavour** [reply]
Thanks for this!

However, I'd love to know how I can load my index page even when offline. My web app was built to run fully offline so it'd be nice to allow my users access it without an internet connection. 

Any idea on how I can do that?

I apologize if my early question didn't make this clear enough.

**Reply by TheAlexLichter** [reply]
If you have a PWA and have it installed + cached, it should "just work"

**Reply by felixfavour** [reply]
I tried using the nuxt/pwa module and I had the same issue so I tried to create my own manifest and service worker myself.  

Do you have any public projects I can have a look at to compare and see what I might be doing wrong?

**Reply by chibx** [reply]
> I tried using the nuxt/pwa module and I had the same issue so I tried to create my own manifest and service worker myself.
> 
> Do you have any public projects I can have a look at to compare and see what I might be doing wrong?

Try https://github.com/elk-zone/elk, it's a nice PWA that some of the nuxt team have been working on

**Reply by felixfavour** [reply]
> > I tried using the nuxt/pwa module and I had the same issue so I tried to create my own manifest and service worker myself.
> > Do you have any public projects I can have a look at to compare and see what I might be doing wrong?
> 
> Try https://github.com/elk-zone/elk, it's a nice PWA that some of the nuxt team have been working on

Thank you, got something else. But will check this out!
